server/s3_get_object_action.cc

```diff
@@ -311,8 +317,14 @@ void S3GetObjectAction::check_full_or_range_object_read() {
     // eg: bytes=0-1024 value
     s3_log(S3_LOG_DEBUG, request_id, "Range found(%s)\n",
            range_header_value.c_str());
-    if (validate_range_header_and_set_read_options(range_header_value)) {
-      next();
+
+    if (S3Option::get_instance()->get_s3_ranged_read_enabled()) {
+      if (validate_range_header_and_set_read_options(range_header_value)) {
+	next();
+      } else {
+	set_s3_error("InvalidRange");
```

Comment:
you can set error code to "OperationNotSupported", since its disabled in option file

Reply:
Done 4 lines below. This case is for ranged read enabled, but when the range validation fails.
Also fixed tabs & spaces (github doesn't work correctly with tabs, see EOS-11827 for the reference).